src/main/java/codeu/controller/LoginServlet.java

```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
 
   /**
    * This function fires when a user submits the login form. It gets the username from the submitted
-   * form data, and then adds it to the session so we know the user is logged in.
+   * form data, checks that they're valid, and either adds the user to the session so we know the 
+   * user is logged in or shows an error to the user. 
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
```

Comment:
Probably worth checking this edge case - what happens if a user doesn't input anything for one of these fields? Does request.getParameter() give null or an empty string? The first possibility could lead to null pointers.

Reply:
getParameter returns a null string if the parameter does not exist, I will add a null check for both parameters to deal with this case